### PR TITLE
doc reader: add cache for scanned elements

### PR DESCRIPTION
**Pull request by larshp**
Add cache for scanned abap doc per class/interface in ZCL_AFF_ABAP_DOC_READER

On my 754 system, it took ~10 seconds to execute all unit tests, this is now down to ~3 seconds

### CODE REVIEW

**Comment by larshp** [comment]
unit test via transpiler is down to ~3 minutes, https://github.com/SAP/abap-file-formats-tools/runs/8250422169?check_suite_focus=true

from ~8 minutes, https://github.com/SAP/abap-file-formats-tools/runs/8228630901?check_suite_focus=true

note that https://github.com/SAP/abap-file-formats-tools/pull/167 will take down the downport, which I guess is most of the time spent in the 3 minutes

File: src/zcl_aff_abap_doc_reader.clas.locals_def.abap [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    tab_comments                 TYPE string_table,
+    column_first_comment         TYPE i,
+    hook_relevant_tok_type       TYPE stokesx,
+    hook_relevant_tok_name       TYPE stokesx,
+    hook_relevant_tok_name_add   TYPE stokesx,
+    hook_relevant_tok_type_stmnt TYPE stokesx,
+    hook_relevant_tok_name_stmnt TYPE stokesx,
+  END OF ty_comment_block.
+TYPES:
+  ty_comment_blocks TYPE STANDARD TABLE OF ty_comment_block WITH EMPTY KEY.
```
Comment (schneidermic0) [inline src/zcl_aff_abap_doc_reader.clas.locals_def.abap:16]: Shouldn't we move these types just to the private section of the global class? As far as I can see it is used only by the global class.

**Review by schneidermic0 — COMMENTED** [review]
Thanks. See my comments

File: src/zcl_aff_abap_doc_reader.clas.locals_def.abap (hunk 1 above)
Comment (larshp) [reply src/zcl_aff_abap_doc_reader.clas.locals_def.abap:16]: used in https://github.com/SAP/abap-file-formats-tools/blob/hvam/cache0809/src/zcl_aff_abap_doc_reader.clas.locals_imp.abap#L24

**Review by schneidermic0 — APPROVED** [review]
LGTM

**Pull request merged**